sns/cm/repair/ut/xform.c

```diff
@@ -479,59 +585,69 @@ static void test_multi_cp_multi_failures(void)
 	 * The test case is illustrated as follows.
 	 * N = 5, K = 2
 	 *
-	 * According to RS encoding,
-	 * if d0 = 'r', d1 = 's', d2 = 't', d3 = 'u', d4 = 'v', then
-	 *    p1 = 'w' and p2 = 'p'
+	 * Calculate parity for below data
+	 * d0 = 'r', d1 = 's', d2 = 't', d3 = 'u', d4 = 'v'
+	 *
 	 * Consider,
 	 * D0 = BUF_NR buffers having data = 'r' (index in ag = 0)
 	 * D1 = FAILED                           (index in ag = 1)
 	 * D2 = BUF_NR buffers having data = 't' (index in ag = 2)
 	 * D3 = BUF_NR buffers having data = 'u' (index in ag = 3)
 	 * D4 = BUF_NR buffers having data = 'v' (index in ag = 4)
-	 * P1 = BUF_NR buffers having data = 'w' (index in ag = 5)
+	 * P1 = BUF_NR buffers having parity     (index in ag = 5)
 	 * P2 = FAILED                           (index in ag = 6)
 	 *
 	 * In above case, after recovery, the accumulator buffers should have
 	 * recovered values of D1 and P2 respectively.
 	 */
-	M0_UT_ASSERT(m0_sns_ir_failure_register(&n_acc_buf[0][0].nb_buffer,
-						        1, &n_rag.rag_ir) == 0);
-	n_rag.rag_fc[0].fc_failed_idx = 1;
-	M0_UT_ASSERT(m0_sns_ir_failure_register(&n_acc_buf[1][0].nb_buffer,
-						        6, &n_rag.rag_ir) == 0);
-	n_rag.rag_fc[0].fc_failed_idx = 6;
```

Comment:
Is this a bug in the original code? It **should** be
```
n_rag.rag_fc[1].fc_failed_idx = 6;
```

But anyway, it is fixed in your patch, right?

Reply:
Yes. In original code 1 and 6 were assigned to `n_rag.rag_fc[0].fc_failed_idx`. I have fixed it in my patch.